packages/flutter/lib/src/material/action_buttons.dart

```diff
@@ -0,0 +1,403 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'action_buttons_theme.dart';
+import 'debug.dart';
+import 'icon_button.dart';
+import 'icons.dart';
+import 'material_localizations.dart';
+import 'scaffold.dart';
+import 'theme.dart';
+
+abstract class _ActionButton extends StatelessWidget {
+  /// Creates a Material Design button.
+  const _ActionButton({
+    super.key,
+    this.color,
+    required this.icon,
+    this.iconSize,
+    required this.onPressed,
+  });
+
+  /// The icon to display inside the button.
+  final Widget icon;
+
+  /// The callback that is called when the button is tapped
+  /// or otherwise activated.
+  ///
+  /// If this is set to null, the button will do a default action
+  /// when it is tapped or activated.
+  final VoidCallback? onPressed;
+
+  /// The color to use for the icon.
+  ///
+  /// Defaults to the [IconThemeData.color] specified in the ambient [IconTheme],
+  /// which usually matches the ambient [Theme]'s [ThemeData.iconTheme].
+  final Color? color;
```

Comment:
IconButtons can (now) be configured with a single ButtonStyle parameter. It would make sense to just provide a ButtonStyle parameter in all of the ActionButton classes (rather than a color parameter) so that developers can change the icon's color in all of the button's states.

Also: if we say that the style parameter will just be passed along to the IconButton, then developers can use IconButton.styleFrom() to easily conjure up simple ButtonStyle instances. That should also preclude the need to explain too much about default computation.

CC FYI @QuncCccccc

Reply:
What should we do about the color in this case https://github.com/flutter/flutter/blob/1220245b330c94ec573d9f4801e93c5c72908f4f/packages/flutter/lib/src/material/icon_button.dart#L626? ButtonStyle will get ignored if theme.useMaterial3 is false.

Reply:
When useMaterial3:true is the default, we should deprecate the BackButton and Close button `close` parameters.
 
 For now, IconButton still has a color parameter, so in this case we can pass the `color` parameter along.

Reply:
Following this, I've added the `color` parameter in the Drawer, and EndDrawer buttons too. I'll remove them if they're not needed.